src/lib/components/artist/header.tsx

```diff
@@ -52,29 +54,35 @@ class Header extends React.Component<HeaderProps, State> {
     })
   }
 
-  handleFollowChange = () => {
+  @eventHandler("Follow Artist")
+  handleFollowChange() { // this now needs to be a normal method, not a property
     const newFollowersCount = this.state.following ? (this.state.followersCount - 1) : (this.state.followersCount + 1)
-    ARTemporaryAPIModule.setFollowArtistStatus(!this.state.following, this.props.artist._id, (error, following) => {
-      if (error) {
-        console.error(error)
-      } else {
-        Events.postEvent(this, {
-          name: following ? "Follow artist" : "Unfollow artist",
-          artist_id: this.props.artist._id,
-          artist_slug: this.props.artist.id,
-          // TODO At some point, this component might be on other screens.
-          source_screen: "artist page",
-        })
-      }
-      this.setState({ following, followersCount: newFollowersCount })
+    return new Promise((resolve, reject) => {
```

Comment:
@l2succes I think you misunderstood, we need these functions to return a promise so that they can signal whether an event succeeded or failed.